lib/active_model_serializers/rspec/matchers/schema.rb

```diff
@@ -0,0 +1,31 @@
+require 'active_model_serializers/test/schema'
+
+module ActiveModelSerializers
+  module RSpecMatchers
+    module Schema
+      extend ActiveSupport::Concern
+
+      included do
+        RSpec::Matchers.define :have_valid_schema do |schema_path|
```

Comment:
Could probably be ( just me typing here, expect errors)

```ruby
      RSpec::Matchers.define :have_valid_schema do |api_schema|
          match do |request_or_response|
            schema_type = request_or_response.respond_to?(:env) && :request
            schema_type ||= request_or_response.respond_to?(:success?) &&  :response
            case schema_type
            when :request
                expect(request_or_response).to be_a_valid_api_request
                assert_request_schema api_schema
            when :response
                expect(request_or_response).to be_a_valid_api_response
                assert_response_schema api_schema
             else 
                 fail UnknownSchemaType.new("#{request_or_response} cannot be identified as a request or response")
             end
          end       
        end
```

alternatively, if we don't want a global matcher

```ruby
included do
    require 'rspec/expectations'
    extend RSpec::Matchers::DSL

    matcher :have_valid_schema do |api_schema|
      match {|request_or_response|
      }
    failure_message {|request_or_response|
    failure_message_when_negated {|request_or_response|
    # etc.
```